main.go

```
@@ -0,0 +1,791 @@
+func loadProgress() *MigrationProgress {
+	progress := &MigrationProgress{
+		CompletedThreads: []int{},
+		FailedThreads:    []int{},
+	}
+
+	data, err := os.ReadFile(ProgressFile)
+	if err != nil {
+		return progress
+	}
+
+	json.Unmarshal(data, progress)
```

golangci-lint (1.64.8)

Error return value of `json.Unmarshal` is not checked

(errcheck)